pages/clubs/form/builder/Form.jsx

```diff
@@ -0,0 +1,397 @@
+var React = require('react');
+
+/**
+
+  A moderately complex form builer for React.
+
+  This Form component converts JSON to HTML forms, using a JSON definition
+  pass in via this.props.fields. This JSON has the following form:
+
+   {
+     fieldname1 : field definition object,
+     fieldname2 : field definition object,
+     ... : ...,
+     ...
+   }
+
+  field definition objects:
+
+   {
+     type: ["text"|"textarea"|choiceGroup"|"checkbox"|checkboxGroup"|ReactComponent],
+     label: string data,
+     placeholder: string data (optional)
+     validator: [instance or array of validator objects],
+     metered: boolean (optional),
+     controller: controller object (optional),
+   }
```

Comment:
That's probably doable - unless we also want something along the lines of "if you don't pass a validator at all, without an error message, we run the default validator with error "You need to fill in {fieldname}". Can we do proptypes for optionals?